src/commands/reset.js

```diff
@@ -8,30 +10,76 @@ class ResetCommand extends BaseCommand {
   /**
    * @param {Object} args
    * @param {Object} flags
-   * @param {DockerCompose} dockerCompose
+   * @param {resetSystemConfig} resetSystemConfig
    * @param {Config} config
+   * @param {ConfigCollection} configCollection
+   * @param {DockerCompose} dockerCompose
+   * @param {Docker} docker
+   * @param {tenderdashInitTask} tenderdashInitTask
    * @return {Promise<void>}
    */
   async runWithDependencies(
     args,
     {
       verbose: isVerbose,
+      hard: isHardReset,
+      'platform-only': isPlatformOnlyReset,
     },
-    dockerCompose,
+    resetSystemConfig,
     config,
+    configCollection,
+    dockerCompose,
+    docker,
+    tenderdashInitTask,
   ) {
     const tasks = new Listr([
+      {
+        title: 'Stop services',
+        task: async () => dockerCompose.stop(config.toEnvs()),
+      },
+      {
+        title: 'Reset platform data',
+        enabled: () => isPlatformOnlyReset,
+        task: () => (
+          new Listr([
+            {
+              title: 'Remove platform services and associated data',
+              task: async () => dockerCompose.rmPlatformOnly(config.toEnvs()),
```

Comment:
DockerCompose is an infrastructure class that shouldn't know about specific business logic. I would recommend moving this function out.

Reply:
Moved this logic to the status command, is that where it should be you think?

Reply:
Yep